src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/IdamServiceImpl.java

```diff
@@ -33,17 +35,17 @@
     public IdamRegistrationInfo registerUser(IdamRegisterUserRequest requestData) {
         IdamRegistrationInfo result;
         try (Response response = idamClient.createUserProfile(requestData)) {
-            ResponseEntity entity = JsonFeignResponseHelper.toResponseEntity(response, Optional.empty());
-            result = new IdamRegistrationInfo(entity.getStatusCode(), Optional.ofNullable(entity));
+            ResponseEntity<Object> entity = JsonFeignResponseHelper.toResponseEntity(response, getResponseMapperClass(response, null));
```

Comment:
may be we can use ObjectUtils.NULL.But why we need hard coded null.

Reply:
ObjectUtils.NULL. cant be passed as param is Class, only accepts null.